### ISSUE TITLE
Disconnection & Automatic Toggle Bugs

### ISSUE THREAD

**Issue by bvelitchkine** [Bastien-and-Gauvain/monorepo#101]
![image](https://github.com/Bastien-and-Gauvain/monorepo/assets/45038783/55acb4a7-ce18-4483-a027-2550c9c44f21)

**Comment by bvelitchkine** [comment]
What causes disconnections is the (yet) unexplained disappearance of the `provider_token` key from the `data.session` object [returned by supabase on oauth](https://github.com/Bastien-and-Gauvain/monorepo/blob/d75ad0885e48104731c76abafec4f6e15dfc5f3b/apps/linkedin-to-notion/src/contents/linkedin-notion-side-panel.tsx#L38-L39) because of this check:

```ts
isLoggedIn={!!session?.provider_token}
```

Here's the payload in the local storage of the extension when it asks for a reconnection:
```json
{
    "access_token": "eyJhbGciOiJIUzI1NiIsImtpZ*****",
    "token_type": "bearer",
    "expires_in": 3600,
    "expires_at": 1707192305,
    "refresh_token": "JfL1*****",
    "user": {...}
}
```

Here's the one in the storage, right after the authentication workflow:
```json
{
    "provider_token": "secret_***",
    "access_token": "eyJhbG***",
    "expires_in": 3600,
    "expires_at": 1707195497,
    "refresh_token": "y1n1Qzw***",
    "token_type": "bearer",
    "user": {...}
}
```

To explain:
1. How does it disappear from the local storage?
2. Why is it happening now and not before? What has changed? (I looked again at the last PR in search of an answer, couldn't see any).

**Comment by GauvainThery** [comment]
Having a look at what we do with sessions and login. IMHO ➡️ absolutely non-sense. We'll have to take the time to fix it. I'm still trying to understand what's best to do.

**Comment by bvelitchkine** [comment]
The log outs were supposedly fixed in #103 but it turns out that they're not. Users keep getting logged out from the extension every hour.
